localpromise.go

```diff
@@ -0,0 +1,70 @@
+package capnp
+
+import (
+	"context"
+)
+
+// ClientHook for a promise that will be resolved to some other capability
+// at some point. Buffers calls in a queue until the promsie is fulfilled,
+// then forwards them.
+type localPromise struct {
+	aq *AnswerQueue
+}
+
+// NewLocalPromise returns a client that will eventually resolve to a capability,
+// supplied via the fulfiller.
+func NewLocalPromise[C ~ClientKind]() (C, Fulfiller[C]) {
+	lp := newLocalPromise()
+	p, f := NewPromisedClient(lp)
+	return C(p), localFulfiller[C]{
+		lp:              lp,
+		clientFulfiller: f,
+	}
+}
+
+func newLocalPromise() localPromise {
+	return localPromise{NewAnswerQueue(Method{})}
```

Comment:
Nit:  linters often complain if you don't use keyed fields in struct literals.  Recommend changing to `localPromise{aq: NewAnswerQueue(Method{})}`.

Reply:
Done.